shared/libraries/config_protocol/include/config_protocol/config_client_component_impl.h

```diff
@@ -193,6 +194,40 @@ void ConfigClientComponentBaseImpl<Impl>::handleRemoteCoreObjectInternal(const C
     ConfigClientPropertyObjectBaseImpl<Impl>::handleRemoteCoreObjectInternal(sender, args);
 }
 
+template <class Impl>
+void ConfigClientComponentBaseImpl<Impl>::remoteUpdateStatuses(const SerializedObjectPtr& serializedStatuses)
+{
+    if (serializedStatuses.hasKey("statuses"))
+    {
+        const auto deserializeContext = createWithImplementation<IComponentDeserializeContext, ConfigProtocolDeserializeContextImpl>(
+            this->clientComm, std::string{}, this->context, nullptr, nullptr, nullptr, nullptr);
+
+        DictPtr<IString, IString> messagesDict;
+        if (serializedStatuses.hasKey("messages"))
+            messagesDict = serializedStatuses.readObject("messages", deserializeContext, nullptr);
+        else
+            messagesDict = Dict<IString, IString>();
+
+        const DictPtr<IString, IEnumeration> statusDict = serializedStatuses.readObject("statuses", deserializeContext, nullptr);
+        const auto statuses = this->statusContainer.getStatuses();
+        const auto statusContainerPrivate = this->statusContainer.template asPtr<IComponentStatusContainerPrivate>(true);
+
+        for (const auto& [name, value] : statusDict)
+        {
+            StringPtr msg;
+            if (messagesDict.hasKey(name))
+                msg = messagesDict.get(name);
+            else
+                msg = String("");
+
+            if (statuses.hasKey(name))
+                statusContainerPrivate.setStatusWithMessage(name, value, msg);
+            else
+                statusContainerPrivate.addStatusWithMessage(name, value, msg);
```

Comment:
I suppose this the general state of our statuses, but do we need to ever remove a status on update?

Reply:
I don't think there is a way to remove the status. There's no method available for this.